### PR TITLE
JFG-1111: add success rate and iterations metrics

### PR DESCRIPTION
**Pull request by anton-antonenko**

### CODE REVIEW

File: src/main/java/com/griddynamics/scenario/jagger/JHttpUserScenario.java [hunk 1]
```diff
@@ -49,6 +47,10 @@ public String getScenarioName() {
         return scenarioName;
     }
 
+    public List<JHttpUserScenarioStep> getUserScenarioSteps() {
+        return userScenarioSteps;
+    }
+
```
Comment (dlatnikov) [inline src/main/java/com/griddynamics/scenario/jagger/JHttpUserScenario.java:53]: in this case we don't need method:  public JHttpUserScenarioStep getUserScenarioStep(int index) {

File: src/main/java/com/griddynamics/scenario/jagger/JHttpUserScenarioInvocationListener.java [hunk 2]
```diff
@@ -73,39 +74,60 @@ public void onStart(InvocationInfo invocationInfo) { }
...
-                            createStepMetricDescription(metricId, metricDisplayName);
+                            String displayName = result.getMetricDisplayName();
+                            createStepLatencyMetricDescription(metricId, displayName);
+                            createStepSuccessRateMetricDescription(metricId + "-sr", displayName);
+                            createIterationsMetricDescription(metricId + "-iterations", displayName);
                         }
-                        getMetricService().saveValue(metricId, value);
+                        getMetricService().saveValue(metricId, result.getLatency());
+                        getMetricService().saveValue(metricId + "-sr", result.getSucceeded() ? 1 : 0);
+                        getMetricService().saveValue(metricId + "-iterations", 1);
```
Comment (dlatnikov) [inline src/main/java/com/griddynamics/scenario/jagger/JHttpUserScenarioInvocationListener.java]: "-sr", "-iterations" - constants?

File: src/main/java/com/griddynamics/scenario/jagger/JHttpUserScenario.java (hunk 1 above)
Comment (anton-antonenko) [reply src/main/java/com/griddynamics/scenario/jagger/JHttpUserScenario.java:53]: Why? It makes obtaining of steps easier and cleaner.

File: src/main/java/com/griddynamics/scenario/jagger/JHttpUserScenarioInvocationListener.java (hunk 2 above)
Comment (anton-antonenko) [reply src/main/java/com/griddynamics/scenario/jagger/JHttpUserScenarioInvocationListener.java]: sure

File: src/main/java/com/griddynamics/scenario/jagger/JHttpUserScenario.java (hunk 1 above)
Comment (dlatnikov) [reply src/main/java/com/griddynamics/scenario/jagger/JHttpUserScenario.java:53]: you can make direct getUserScenarioSteps().get(index)

Comment (anton-antonenko) [reply src/main/java/com/griddynamics/scenario/jagger/JHttpUserScenario.java:53]: yes, but getUserScenarioStep(int index) takes less characters) it looks better

**Review by dlatnikov — APPROVED** [review]

**Pull request merged**